File: tests/zero_lag_exponential_moving_average_test.py
import numpy as np

from taflow import ZeroLagExponentialMovingAverage


def test_lifecycle() -> None:
    values = np.linspace(1.0, 20.0, 32)
    state = ZeroLagExponentialMovingAverage(9).extend(values)
    assert len(state) == len(values)
    assert state.reset().value is None

File: crates/taflow-core/src/stream/zero_lag_exponential_moving_average_test.rs
use super::zero_lag_exponential_moving_average::ZeroLagExponentialMovingAverage;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = ZeroLagExponentialMovingAverage::new(5).unwrap();
    for value in 0..8 {
        state.append(value as f64);
    }
    assert!(state.value().is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
